keras_cv/layers/preprocessing/random_flip_test.py

```diff
@@ -35,12 +35,12 @@ def _run_test(self, mode, expected_output=None, mock_random=None):
                 expected_output = np.flip(expected_output, axis=2)
             if mode == "vertical" or mode == "horizontal_and_vertical":
                 expected_output = np.flip(expected_output, axis=1)
+        layer = RandomFlip(mode)
         with tf.compat.v1.test.mock.patch.object(
-            tf.random,
-            "uniform",
+            layer._random_generator,
```

Comment:
does this alias here cause issues?  Currently, we have no compat references in the code base.  I'd like to avoid introducing this if we can.  

@qlzh727 what is the most modern way to mock random behavior?

Reply:
@LukeWood  I am unable to use the tensor to check the bool value - ```OperatorNotAllowedInGraphError: Using a symbolic `tf.Tensor` as a Python `bool` is not allowed in Graph execution```
I tried tf.cond() to assign random True/False. But I am still seeing the sam error when the result is checked in if conditions in augment_*

Reply:
> does this alias here cause issues? Currently, we have no compat references in the code base. I'd like to avoid introducing this if we can.
> 
> @qlzh727 what is the most modern way to mock random behavior?

https://github.com/tensorflow/tensorflow/blob/5ad4301046bf785abbc7ff9ff474df523781d4a6/tensorflow/python/platform/test.py#L38:L41


https://github.com/keras-team/keras/blob/master/keras/layers/preprocessing/image_preprocessing_test.py#L412

Reply:
> @LukeWood, I dont think the issue is with `mock`. I am unable to use the tensor to check the bool value - `` OperatorNotAllowedInGraphError: Using a symbolic `tf.Tensor` as a Python `bool` is not allowed in Graph execution `` I tried `tf.cond(self._random_generator.random_uniform(shape=[]) > 0.5, lambda: True, lambda: False)` to assign random True/False. But I am still seeing the same error when the result is checked in if conditions in augment_*. what is the best way to handle bool tensor here?

Regardless of the root cause, I'd like to avoid using a compat alias in the package if we can.   The issue is with using the boolean directly in an if statement, so what you can do instead is use a tf.cond to apply the horizontal/vertical flip.  For example:

```
 if transformation["flip_vertical"]:
            bounding_boxes_out = tf.stack(
                [
                    bounding_boxes_out[:, 0],
                    h - bounding_boxes_out[:, 3],
                    bounding_boxes_out[:, 2],
                    h - bounding_boxes_out[:, 1],
                ],
                axis=-1,
            )
```

could become:

```
def flip_vertical():
    bounding_boxes_out = tf.stack(
        [
            bounding_boxes_out[:, 0],
            h - bounding_boxes_out[:, 3],
            bounding_boxes_out[:, 2],
            h - bounding_boxes_out[:, 1],
        ],
        axis=-1,
    )
bounding_boxes = tf.cond(transformation["flip_vertical"], lambda: flip_vertical, bounding_boxes)
```

or something similar.

Either way I'd like to remove the v1.compat.patch usage.

Reply:
done